src/mod_bosh.erl

```diff
@@ -89,7 +89,7 @@ start(_HostType, Opts) ->
         false ->
             mod_bosh_backend:start(Opts),
             {ok, _Pid} = mod_bosh_socket:start_supervisor(),
-            ejabberd_hooks:add(node_cleanup, global, ?MODULE, node_cleanup, 50)
+            gen_hook:add_handler(node_cleanup, global, fun ?MODULE:node_cleanup/3, #{}, 50)
```

Comment:
This hook should be unregistered when the module is unloaded. I recommend to move it to `hooks/1` and add the following code at start/stop functions:
```
gen_hook:add_handlers(hooks(HostType)) % start
gen_hook:delete_handlers(hooks(HostType)) % stop
```

The specification for `hooks/1`: `-spec hooks(mongooseim:host_type()) -> gen_hook:hook_list().`.

Reply:
done 👍